redash/tasks/queries/execution.py

```diff
@@ -43,16 +43,24 @@ def enqueue_query(
             job_id = pipe.get(_job_lock_id(query_hash, data_source.id))
             if job_id:
                 logger.info("[%s] Found existing job: %s", query_hash, job_id)
+                job_exists = Job.exists(job_id)
+                job_complete = None
+
+                if job_exists:
+                    job = Job.fetch(job_id)
```

Comment:
While it's not very likely, it still can happen that between the call to `exists` to `fetch`, the job will be removed. Is there a version of `fetch` that returns `None` if job doesn't exist? Or maybe just catch exception?

Reply:
24ca387